tests/lib/cli-engine/cascading-config-array-factory.js

```diff
@@ -575,6 +575,24 @@ describe("CascadingConfigArrayFactory", () => {
             });
 
 
+            it("should resolve plugins/parsers nested within a shared package", () => {
+                const factory = new CascadingConfigArrayFactory({
+                    cwd: getFixturePath("nested-plugins-parser")
+                });
+                const file = getFixturePath("nested-plugins-parser", "foo.js");
+                const expected = {
+                    parser: getFixturePath("nested-plugins-parser", "node_modules", "eslint-config-awesome", "node_modules", "awesome-parser", "index.js"),
+                    plugins: ["awesome"],
+                    rules: {
+                        "awesome/is-awesome": [2]
+                    }
+                };
+                const actual = getConfig(factory, file);
```

Comment:
This throws with:

```
     Error: Failed to load plugin 'awesome' declared in '.eslintrc.yml » eslint-config-awesome': Cannot find module 'eslint-plugin-awesome'
Require stack:
- /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/__placeholder__.js
      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
      at Function.resolve (internal/modules/cjs/helpers.js:32:19)
      at Object.resolve (lib/shared/relative-module-resolver.js:44:50)
      at ConfigArrayFactory._loadPlugin (lib/cli-engine/config-array-factory.js:840:39)
      at names.reduce (lib/cli-engine/config-array-factory.js:729:33)
      at Array.reduce (<anonymous>)
      at ConfigArrayFactory._loadPlugins (lib/cli-engine/config-array-factory.js:725:22)
      at ConfigArrayFactory._normalizeObjectConfigDataBody (lib/cli-engine/config-array-factory.js:551:32)
      at _normalizeObjectConfigDataBody.next (<anonymous>)
      at ConfigArrayFactory._normalizeObjectConfigData (lib/cli-engine/config-array-factory.js:490:20)
      at _normalizeObjectConfigData.next (<anonymous>)
      at ConfigArrayFactory._normalizeObjectConfigDataBody (lib/cli-engine/config-array-factory.js:544:25)
      at _normalizeObjectConfigDataBody.next (<anonymous>)
      at ConfigArrayFactory._normalizeObjectConfigData (lib/cli-engine/config-array-factory.js:490:20)
      at _normalizeObjectConfigData.next (<anonymous>)
      at createConfigArray (lib/cli-engine/config-array-factory.js:306:25)
      at ConfigArrayFactory.loadInDirectory (lib/cli-engine/config-array-factory.js:399:16)
      at CascadingConfigArrayFactory._loadConfigInAncestors (lib/cli-engine/cascading-config-array-factory.js:293:46)
      at CascadingConfigArrayFactory.getConfigArrayForFile (lib/cli-engine/cascading-config-array-factory.js:238:18)
      at getConfig (tests/lib/cli-engine/cascading-config-array-factory.js:144:22)
      at Context.it (tests/lib/cli-engine/cascading-config-array-factory.js:590:32)
```

Reply:
Debug output leading up to the failure:

```
  eslint:config-array-factory Loading YAML config file: /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/.eslintrc.yml +0ms
  eslint:config-array-factory Config file found: /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/.eslintrc.yml +4ms
  eslint:config-array-factory Loading {extends:"awesome"} relative to /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/.eslintrc.yml +14ms
  eslint:config-array-factory package.json was not found: Cannot find module 'eslint-config-awesome/package.json'
Require stack:
- /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/.eslintrc.yml +1ms
  eslint:config-array-factory Loaded: eslint-config-awesome (/private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/node_modules/eslint-config-awesome/index.js) +0ms
  eslint:config-array-factory Loading JS config file: /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/node_modules/eslint-config-awesome/index.js +1ms
  eslint:config-array-factory Loading parser "awesome-parser" from /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/node_modules/eslint-config-awesome/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module 'awesome-parser/package.json'
Require stack:
- /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/node_modules/eslint-config-awesome/index.js +1ms
  eslint:config-array-factory Loaded: awesome-parser (/private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/node_modules/eslint-config-awesome/node_modules/awesome-parser/index.js) +0ms
  eslint:config-array-factory Loading plugin "awesome" from /private/var/folders/j6/0rhhpj0x0db7yqhhby7x27_r0000gn/T/eslint/fixtures/config-hierarchy/nested-plugins-parser/node_modules/eslint-config-awesome/index.js +1ms
  eslint:config-array-factory Failed to load plugin 'awesome' declared in '.eslintrc.yml » eslint-config-awesome'. +0ms
```

Reply:
Note that the ConfigArrayFactory is trying to resolve plugin 'awesome' relative to the root package, not eslint-config-awesome (but the require stack thinks otherwise)